partiql-eval/src/main/kotlin/org/partiql/eval/PartiQLResult.kt

```diff
@@ -1,12 +1,12 @@
 package org.partiql.eval
 
-import org.partiql.value.PartiQLValue
+import org.partiql.value.PartiQLData
 import org.partiql.value.PartiQLValueExperimental
 
 public sealed interface PartiQLResult {
 
     @OptIn(PartiQLValueExperimental::class)
-    public data class Value(public val value: PartiQLValue) : PartiQLResult
+    public data class Value(public val value: PartiQLData) : PartiQLResult
```

Comment:
We should probably have a little doc on naming now. The Value/value and type not being a PartiQLValue is interesting, but perhaps to get the diff small. I do think we should incorporate "cursor" (a'la SQL) or "reader" (a'la Ion) such as

```
public data class Value(public val cursor: ResultCursor) : PartiQLResult
public data class Value(public val reader: PartiQLReader) : PartiQLResult
```

PartiQLReader / PartiQLCursor / PartiQLData / etc etc

Reply:
Updated the PartiQLData to be PartiQLCursor. Yeah, I opted to not rename `PartiQLResult.Value` to something else (maybe `PartiQLResult.Data` or `PartiQLResult.Cursor`) due to the large diff that would occur. That can be a follow-up to this PR.